code/modules/admin/DB ban/functions.dm

```diff
@@ -330,12 +333,20 @@ datum/admins/proc/DB_ban_unban_by_id(var/id)
 	var/unban_ckey = src.owner:ckey
 	var/unban_computerid = src.owner:computer_id
 	var/unban_ip = src.owner:address
+	
+	
+	if(!challenge_phrase)
+		var/sql_update = "UPDATE [format_table_name("ban")] SET unbanned = 1, unbanned_datetime = Now(), unbanned_ckey = '[unban_ckey]', unbanned_computerid = '[unban_computerid]', unbanned_ip = '[unban_ip]' WHERE id = [id]"
+		message_admins("[key_name_admin(usr)] has lifted [pckey]'s ban.",1)
+		var/DBQuery/query_update = dbcon.NewQuery(sql_update)
+		query_update.Execute()
+	else
+		var/sql_update = "UPDATE [format_table_name("ban")] SET unbanned = 1, unbanned_datetime = Now(), unbanned_ckey = '[unban_ckey]', unbanned_computerid = '[unban_computerid]', unbanned_ip = '[unban_ip]', challenge = 1 WHERE id = [id]"

```

Comment:
This is going to require me to alter my definition of what counts as 'banned', because unbanned=1 no longer actually means the user is fully unbanned (they could still be required to enter the challenge phrase).

Does the ban panel's reporting of ban status show 'pending challenge phrase' unbans distinctly from truly 'unbanned' bans?